internal/slicer/slicer.go

```diff
@@ -283,28 +322,38 @@ func Run(options *RunOptions) error {
 				targetPaths = []string{targetPath}
 			}
 			for _, targetPath := range targetPaths {
+				if strings.HasSuffix(targetPath, "/") {
+					untilDirs = append(untilDirs, targetPath)
+					continue
+				}
 				realPath, err := content.RealPath(targetPath, scripts.CheckRead)
 				if err == nil {
-					if strings.HasSuffix(targetPath, "/") {
-						untilDirs = append(untilDirs, realPath)
-					} else {
-						err = os.Remove(realPath)
-					}
+					err = os.Remove(realPath)
 				}
 				if err != nil {
 					return fmt.Errorf("cannot perform 'until' removal: %w", err)
 				}
+				pathTrack.removeTarget(targetPath)
 			}
 		}
 	}
-	for _, realPath := range untilDirs {
-		err := os.Remove(realPath)
-		// The non-empty directory error is caught by IsExist as well.
-		if err != nil && !os.IsExist(err) {
+	for _, targetPath := range untilDirs {
+		realPath, err := content.RealPath(targetPath, scripts.CheckRead)
+		if err == nil {
+			err = os.Remove(realPath)
+		}
+		if err == nil {
+			pathTrack.removeTarget(targetPath)
+		} else if !os.IsExist(err) {
+			// The non-empty directory error is caught by IsExist as well.
 			return fmt.Errorf("cannot perform 'until' removal: %#v", err)
```

Comment:
Perhaps:
```suggestion
			return fmt.Errorf("cannot perform 'until' removal: %w", err)
```
Like in the code above (: